packages/front-end/components/Modal.tsx

```diff
@@ -20,7 +20,7 @@ type ModalProps = {
   autoCloseOnSubmit?: boolean;
   solidOverlay?: boolean;
   close?: () => void;
-  submit?: () => Promise<void>;
+  submit?: () => void; //TODO: Make sure removing the Promise here doesn't introduce regressions.
```

Comment:
I'd love a second look at this change to make sure I'm not introducing any regressions.